sw/airborne/firmwares/rotorcraft/stabilization/stabilization_rate_indi.c

```diff
@@ -23,37 +23,140 @@
  *  Rate stabilization for rotorcrafts based on INDI by Ewoud Smeur.
  */
 #include "firmwares/rotorcraft/stabilization.h"
-#include "firmwares/rotorcraft/stabilization/stabilization_rate.h"
+#include "firmwares/rotorcraft/stabilization/stabilization_rate_indi.h"
```

Comment:
So, this where the magic should come from. In fact `rate_indi` must implement the `rate` interface (and in this case add extra function for external control).
I'm suggesting the following structure:
- in `stabilization_rate_indi.h`
  - include `stabilization_rate.h`
  - remove the basic `stabilization_rate_*` functions declarations, already provided by rate.h
  - keep the `stabilization_rate_indi_*` for external modules
- in `stabilization_rate_indi.c`
  - include `stabilization_rate_indi.h` as you did now
  - implement the `rate_indi_*` functions as you did now
  - implement the basic `rate_*` functions by calling the proper indi related functions, especially the `stabilization_rate_run` function
- in `stabilization_rate_indi.xml` module file:
  - change the header to include `stabilization_rate_indi.h`
  - compile only the `stabilization_rate_indi.c` as it is now

How is it going to work ?
- if you are in rate mode, the autopilot will call the basic `stabilization_rate_*` functions, setting the setpoint from RC and calling the run afterwards
- if you are in a custom mode, you have to set the setpoint and then call the `rate_indi_enter` and `rate_indi_run` functions by yourself

Does it makes sense ?

Reply:
Yes, I think I got it. Thanks a lot for your detailed feedback! 
I got through this morning and push the code in a few minutes. Though I have been doing without compiling it, because there is something I would like to review further in this exact file.